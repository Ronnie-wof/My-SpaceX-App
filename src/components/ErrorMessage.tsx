import React from 'react'
import {View, StyleSheet, Text} from 'react-native'
import { Entypo } from '@expo/vector-icons';


interface errorMessageProps {
    message: string
}
const ErrorMessage:React.FC<errorMessageProps> = ({message}) => {
    return (
        <View style={styles.container}>
            <Entypo name="emoji-sad" size={24} color="black" />
            <Text style={styles.error}>{message}</Text>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#fff',
      alignItems: 'center',
      justifyContent: 'center',
    },
    error: {
        fontSize:20,
        color: '#191970'
    }
  });

  export default ErrorMessage;