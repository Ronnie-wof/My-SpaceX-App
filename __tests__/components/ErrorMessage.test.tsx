// tests the ErrorMessage component
import React from 'react';
import renderer from 'react-test-renderer';

import ErrorMessage from '../../src/components/ErrorMessage';

describe('ErrorMessage component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ErrorMessage message="An error occurred" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
