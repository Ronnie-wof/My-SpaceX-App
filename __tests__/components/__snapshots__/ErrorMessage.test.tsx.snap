// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text />
  <Text
    style={
      {
        "color": "#191970",
        "fontSize": 20,
      }
    }
  >
    An error occurred
  </Text>
</View>
`;
